### PR TITLE
test: add packet-forwarding middleware e2e test

### PR DESCRIPTION
**Pull request by pirtleshell**
## Description
Adds a first pass PoC of using interchaintest for testing more complex IBC flows.

Currently hardcoded to v0.26.0, `make test-ibc` will spin up a network with the following:
* 2 validators & full archive node for each binary: kava, cosmoshub, & osmosis
* an ibc relayer
* IBC channels opened between `kava <-> cosmoshub` & `kava <-> osmosis`

Tests the following:
* simple IBC transfer of kava to cosmoshub
* packet-forwarding middleware test that IBC transfers kava from cosmoshub to osmosis, forwarded through kava

Includes a CI step that runs the above.

## Next steps
* build interchaintest-compatible container from current branch code & use in `make test-ibc`
* configure evmutil params & evm state in network genesis to have internal bridge-configured erc20 (for ibc testing of USDT)

### CODE REVIEW

File: tests/e2e-ibc/main_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+		tx, err := gaia.SendIBCTransfer(ctx, gaiaToKavaChannelID, gaiaUser.KeyName(), transfer, ibc.TransferOptions{
+			// packet forwarding middleware!
+			Memo: fmt.Sprintf(`{
+	"forward": {
+		"receiver": "%s",
+		"port": "transfer",
+		"channel": "%s"
+	}
+}`, dstAddress, kavaToOsmoChannelID),
```
Comment (pirtleshell) [inline tests/e2e-ibc/main_test.go:185]: ✨ here is the IBC transfer from cosmoshub -> osmosis passing through kava

File: tests/e2e-ibc/main_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				Denom:          "ukava",
+				GasPrices:      "0ukava", // 0 gas price makes calculating expected balances simpler
+				GasAdjustment:  1.5,
+				TrustingPeriod: "168h0m0s",
+				// ModifyGenesis:  cosmos.ModifyGenesis(genesis), // TODO: configure evm things for internal bridge testing
+				// CoinType: "60", // might need this to sign evm txs. will need to override decimals to be 6 again.
+			},
+		},
+		{Name: "gaia", Version: "v15.2.0", ChainConfig: ibc.ChainConfig{GasPrices: "0.0uatom"}},
+		{Name: "osmosis", Version: "v24.0.1"},
```
Comment (pirtleshell) [inline tests/e2e-ibc/main_test.go:43]: we're using the actual chain binaries for mainnet cosmoshub and osmosis 🤘

File: tests/e2e-ibc/main_test.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	ctx := context.Background()
+
+	// setup chains
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			Name: "kava",
+			ChainConfig: ibc.ChainConfig{
+				Type:           "cosmos",
+				ChainID:        "kava_8888-1",
+				Images:         []ibc.DockerImage{{Repository: "ghcr.io/strangelove-ventures/heighliner/kava", Version: "v0.26.0", UidGid: "1025:1025"}},
```
Comment (pirtleshell) [inline tests/e2e-ibc/main_test.go:31]: here is the hardcoded v0.26.0 kava image we'll want to build from current branch code for CI

File: tests/e2e-ibc/go.mod [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	github.com/ChainSafe/go-schnorrkel => github.com/ChainSafe/go-schnorrkel v0.0.0-20200405005733-88cbf1b4c40d
+	github.com/ChainSafe/go-schnorrkel/1 => github.com/ChainSafe/go-schnorrkel v1.0.0
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	github.com/vedhavyas/go-subkey => github.com/strangelove-ventures/go-subkey v1.0.7
+)
+
+require (
+	cosmossdk.io/math v1.3.0
+	github.com/cosmos/ibc-go/v8 v8.1.0
+	github.com/strangelove-ventures/interchaintest/v8 v8.2.0
```
Comment (nddeluca) [inline tests/e2e-ibc/go.mod:15]: Nit: Should these be matched with kava v0.26.x versions?

File: tests/e2e-ibc/go.mod [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+	modernc.org/opt v0.1.3 // indirect
+	modernc.org/sqlite v1.28.0 // indirect
+	modernc.org/strutil v1.1.3 // indirect
+	modernc.org/token v1.0.1 // indirect
+	nhooyr.io/websocket v1.8.7 // indirect
+	pgregory.net/rapid v1.1.0 // indirect
+	sigs.k8s.io/yaml v1.4.0 // indirect
+)
+
+// replace github.com/strangelove-ventures/interchaintest/v8 => ../../../interchaintest
```
Comment (nddeluca) [inline tests/e2e-ibc/go.mod:263]: Probably can remove this comment -- rare for another dev to have the same relative dir setup and best left to a go.work file

**Review by nddeluca — APPROVED** [review]

File: tests/e2e-ibc/go.mod (hunk 5 above)
Comment (pirtleshell) [reply tests/e2e-ibc/go.mod:263]: 👍 go.work struggles because of mismatched versions and overrides in this test suite vs kava.
but will remove

Comment (nddeluca) [reply tests/e2e-ibc/go.mod]: Ah I see now, go.work isn't working with latest master or release/v0.26.x

File: tests/e2e-ibc/go.mod (hunk 4 above)
Comment (pirtleshell) [reply tests/e2e-ibc/go.mod:15]: as you mentioned, i think we'll want them to match for conformance testing. i'll note it in a ticket to be updated in a future pr 👍

**Pull request merged**